### PR TITLE
PERF: Fix performance regression in get_loc of IntervalIndex

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

caused in https://github.com/pandas-dev/pandas/commit/184e1674

No need to call validate here

https://asv-runner.github.io/asv-collection/pandas/#indexing.IntervalIndexing.time_loc_list

### CODE REVIEW

File: pandas/core/arrays/interval.py [hunk 1]
```diff
@@ -727,7 +730,7 @@ def __getitem__(
         if np.ndim(left) > 1:
             # GH#30588 multi-dimensional indexer disallowed
             raise ValueError("multi-dimensional indexing not allowed")
-        return self._shallow_copy(left, right)
+        return self._shallow_copy(left, right, verify_integrity=False)
```
Comment (jbrockmendel) [inline pandas/core/arrays/interval.py:733]: couldnt we also skip _ensure_simple_new_inputs, i.e. just do

`return self._simple_new(left, right, dtype=self.dtype)`

File: pandas/core/arrays/interval.py [hunk 2]
```diff
@@ -675,7 +677,8 @@ def _shallow_copy(self: IntervalArrayT, left, right) -> IntervalArrayT:
         """
         dtype = IntervalDtype(left.dtype, closed=self.closed)
         left, right, dtype = self._ensure_simple_new_inputs(left, right, dtype=dtype)
-        self._validate(left, right, dtype=dtype)
+        if verify_integrity:
```
Comment (jbrockmendel) [inline pandas/core/arrays/interval.py:680]: so is this still necessary?

File: pandas/core/arrays/interval.py (hunk 1 above)
Comment (phofl) [reply pandas/core/arrays/interval.py]: let's see if it works, probably not necessary

File: pandas/core/arrays/interval.py (hunk 2 above)
Comment (phofl) [reply pandas/core/arrays/interval.py]: Not for this regression, but I think we should look through all usages to see if it's necessary everywhere

**Comment by jbrockmendel** [comment]
pandas/core/arrays/interval.py:729: error: Argument 2 to "_simple_new" of "IntervalArray" has incompatible type "Union[Period, Timestamp, Timedelta, NaTType, DatetimeArray, TimedeltaArray, ndarray[Any, Any]]"; expected "Union[Union[DatetimeArray, TimedeltaArray], ndarray[Any, Any]]"  [arg-type]

**Review by jbrockmendel — APPROVED** [review]
LGTM pending green

**Pull request merged**